app/admin/applications/page.tsx

```diff
@@ -0,0 +1,17 @@
+'use client';
+import Custom404 from '@/app/not-found';
+import { useSession } from 'next-auth/react';
+
+const ApplicationsPage = () => {
+  const { data: session } = useSession();
+
+  if (!session || session.user?.role != 'admin') return <Custom404 />;
```

Comment:
Denne sjekken burde kunne gjøres kun ett sted, i `layout.tsx` for eksempel. Istedenfor å sjekke på hver underside.

Reply:
Dersom jeg leger det til i layout vil ikke da ikke admin brukere ikke få tilgang til resten av fondssiden?

Reply:
Du kan lage en layout i admin-mappen, så vil det kun bli sjekket på /admin-sider tror jeg.